packages/build-tools/src/steps/functions/repack.ts

```diff
@@ -0,0 +1,174 @@
+import fs from 'node:fs';
+import os from 'node:os';
+import path from 'node:path';
+
+import { Platform } from '@expo/eas-build-job';
+import { type bunyan } from '@expo/logger';
+import {
+  BuildFunction,
+  BuildStepInput,
+  BuildStepInputValueTypeName,
+  BuildStepOutput,
+  spawnAsync,
+} from '@expo/steps';
+import {
+  repackAppAndroidAsync,
+  repackAppIosAsync,
+  type Logger,
+  type SpawnProcessAsync,
+  type SpawnProcessOptions,
+  type SpawnProcessPromise,
+  type SpawnProcessResult,
+} from '@expo/repack-app';
+
+export function createRepackBuildFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'repack',
+    name: 'Repack app',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'source_app_path',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: true,
+      }),
+      BuildStepInput.createProvider({
+        id: 'platform',
+        required: false,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+      BuildStepInput.createProvider({
+        id: 'output_path',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: false,
+      }),
+      BuildStepInput.createProvider({
+        id: 'embed_bundle_assets',
+        allowedValueTypeName: BuildStepInputValueTypeName.BOOLEAN,
+        required: false,
+      }),
+    ],
+    outputProviders: [
+      BuildStepOutput.createProvider({
+        id: 'output_path',
+        required: true,
+      }),
+    ],
+    fn: async (stepsCtx, { inputs, outputs, env }) => {
+      const projectRoot = stepsCtx.workingDirectory;
+      const verbose = stepsCtx.global.env['EAS_VERBOSE'] === '1';
+
+      const platform =
+        (inputs.platform.value as Platform) ?? stepsCtx.global.staticContext.job.platform;
+      if (![Platform.ANDROID, Platform.IOS].includes(platform)) {
+        throw new Error(
+          `Unsupported platform: ${platform}. Platform must be "${Platform.ANDROID}" or "${Platform.IOS}"`
+        );
+      }
+
+      const repackLogger = createBunyanLoggerAdapter(stepsCtx.logger);
+      const repackSpawnAsync = createSpawnAsyncStepAdapter(verbose, stepsCtx.logger);
+
+      const tmpDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), `repack-`));
+      const workingDirectory = path.join(tmpDir, 'working-directory');
+      await fs.promises.mkdir(workingDirectory, { recursive: true });
```

Comment:
```suggestion
      await fs.promises.mkdir(workingDirectory);
```
If I'm not mistaken `mkdtemp` should already take care of creating `tmpDir`.